__tests__/app.test.js

```diff
@@ -179,6 +179,66 @@ describe("GET /api/reviews/:review_id endpoint.", () => {
 
 })
 
+describe("GET /api/reviews/:review_id/comments endpoint.", () => {
+
+    describe("Functionality tests:", () => {
+
+        test("Returns an array of comments for the specified review, each containing the correct six properties.", () => {
+            return request(app).get('/api/reviews/2/comments')
+            .expect(200)
+            .then(({body}) => {
+                expect(body.comments.length).toBe(3);
+                body.comments.forEach(comment => {
+                    expect(comment.comment_id).toEqual(expect.any(Number));
+                    expect(comment.votes).toEqual(expect.any(Number));
+                    expect(comment.created_at).toEqual(expect.any(String));
+                    expect(comment.author).toEqual(expect.any(String));
+                    expect(comment.body).toEqual(expect.any(String));
+                    expect(comment.review_id).toEqual(expect.any(Number));
+                })
+            })
+        })
+    
+    })
+
+    describe("Error handling tests:", () => {
+
+        test("Responds to invalid review comments path with 404 status and 'That page does not exist' message.", () => {
+            return request(app).get('/api/reviews/2/comment')
+            .expect(404)
+            .then(({body}) => {
+                expect(body.message).toBe("That page does not exist.");
+            })
+        })
+
+        test("Responds to valid but non-existent Review ID with 404 status and 'That review either has no comments or does not exist' message.", () => {
+            return request(app).get('/api/reviews/1000/comments')
+            .expect(404)
+            .then(({body}) => {
+                expect(body.message).toBe("That review either has no comments or does not exist.");
+            })
+        })
+
+        test("If review has no comments, returns 404 status and 'That review either has no comments or does not exist' message.", () => {
```

Comment:
If the review does exist - we don't want to be sending back a 404 and an error message. It's a perfectly valid article so we should send back an empty array instead of an error message

Reply:
Cheers - I did wonder. Corrected now using separate checkReviewExists function.